### PR TITLE
Fix source tests on Windows

### PR DESCRIPTION
**Pull request by omichel**
The header test was broken (parsing files inside .git folder).
The cppcheck test was broken (on Windows, it relies on a newer version of cppcheck which detects more problems)

### CODE REVIEW

**Comment by omichel** [comment]
:+1:

**Comment by DavidMansolino** [comment]
:eyes:

File: src/webots/nodes/WbCompass.hpp [hunk 1]
```diff
@@ -26,7 +26,7 @@ class WbCompass : public WbSolidDevice {
 public:
   // constructors and destructor
   explicit WbCompass(WbTokenizer *tokenizer = NULL);
-  WbCompass(const WbCompass &other);
+  explicit WbCompass(const WbCompass &other);
```
Comment (DavidMansolino) [inline src/webots/nodes/WbCompass.hpp]: Why here you define this constructor as `explicit` and not in the other class (e.g. `WbBrake.hpp` just before)?

File: tests/sources/test_header_version.py [hunk 2]
```diff
@@ -42,6 +43,7 @@ def setUp(self):
         # 2. Get all the PROTO files
         self.files = []
         for rootPath, dirNames, fileNames in os.walk(os.environ['WEBOTS_HOME']):
+            dirNames[:] = [d for d in dirNames if d not in skippedDirectories]
```
Comment (DavidMansolino) [inline tests/sources/test_header_version.py]: Nice Pythonic way of doing the filtering !

File: src/webots/nodes/WbCompass.hpp (hunk 1 above)
Comment (omichel) [reply src/webots/nodes/WbCompass.hpp]: Because otherwise cppcheck will complain about it. Strangely it doesn't complain about it in other classes...

Comment (DavidMansolino) [reply src/webots/nodes/WbCompass.hpp]: Ok, thank you for the explanation, this sounds like a cppcheck bug.

**Comment by omichel** [comment]
👍

**Comment by omichel** [comment]
👍

File: tests/sources/test_clang_format.py [hunk 3]
```diff
@@ -93,6 +93,27 @@ def test_sources_are_clang_format_compliant(self):
...
+                    if not found:
+                        continue
+                    found = False
+                    for directory in skippedPaths:
+                        if line.startswith(directory):
+                            found = True
+                            break
+                    if found:
+                        continue
+                    found = False
```
Comment (DavidMansolino) [inline tests/sources/test_clang_format.py]:
```suggestion
```

File: tests/sources/test_clang_format.py [hunk 4]
```diff
@@ -93,6 +93,27 @@ def test_sources_are_clang_format_compliant(self):
...
                     if extension not in extensions:
                         continue
+                    found = False
+                    for directory in directories:
+                        if line.startswith(directory):
+                            found = True
+                            break
+                    if not found:
+                        continue
+                    found = False
```
Comment (DavidMansolino) [inline tests/sources/test_clang_format.py]:
```suggestion
```

**Comment by omichel** [comment]
👍

**Comment by omichel** [comment]
I could fix a bug introduced in this PR thanks to the unwanted appveyor test (Webots compilation). 😄

👍

**Comment by omichel** [comment]
The travis API tests have apparently several timeout failures 😒...

**Comment by DavidMansolino** [comment]
There was many test suite failure (not only failure):
```
### api test

OK: accelerometer
OK: asymmetric_friction
FAILURE with battery: The position value measured by the position sensor should be 0.5739 and not nan
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/battery.wbt: Timeout the results file has not been written quickly enough
FAILURE with brake: wb_brake_get_position_sensor didn't work correctly.
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/brake.wbt: Timeout the results file has not been written quickly enough
OK: camera_color
OK: camera_color_focus
OK: camera_color_motion_blur
OK: camera_color_spherical
OK: camera_image_update
OK: camera_noise_mask
OK: camera_recognition
OK: camera_revert
OK: center_of_mass_and_contact_points
OK: compass
OK: device
OK: display
OK: display_attach_camera
OK: display_infrared
OK: distance_sensor_enable_disable
OK: distance_sensor_infra-red
OK: distance_sensor_laser
OK: distance_sensor_resolution
OK: distance_sensor_sonar
OK: emitter_receiver_determinism_supervisor
OK: emitter_receiver_enable_disable_serial_emitter_radio_receiver
OK: emitter_receiver_enable_disable_infra_red_emitter
OK: emitter_receiver_enable_disable_radio_emitter
OK: emitter_receiver_enable_disable_serial_receiver
OK: emitter_receiver_enable_disable_infra_red_receiver
OK: emitter_receiver_infra-red_emitter_0_receiver_0
OK: emitter_receiver_infra-red_emitter_1
OK: emitter_receiver_infra-red_receiver_2
OK: emitter_receiver_infra-red_receiver_3
OK: emitter_receiver_infra-red_aperture_emitter_0
OK: emitter_receiver_infra-red_aperture_emitter_1
OK: emitter_receiver_infra-red_aperture_receiver_2
OK: emitter_receiver_infra-red_aperture_receiver_3
OK: emitter_receiver_invalid_function_call_emitter
OK: emitter_receiver_invalid_function_call_receiver
OK: emitter_receiver_physics_plugin
OK: emitter_receiver_radio_emitter_0
OK: emitter_receiver_radio_emitter_1
OK: emitter_receiver_radio_receiver_2
OK: emitter_receiver_radio_receiver_3
OK: emitter_receiver_radio_aperture_emitter_0
OK: emitter_receiver_radio_aperture_emitter_1
OK: emitter_receiver_radio_aperture_receiver_2
OK: emitter_receiver_radio_aperture_receiver_3
OK: emitter_receiver_serial_emitter_0
OK: emitter_receiver_serial_emitter_1
OK: emitter_receiver_serial_emitter_2
OK: emitter_receiver_serial_receiver_3
OK: emitter_receiver_serial_aperture_emitter_0
OK: emitter_receiver_serial_aperture_emitter_1
OK: emitter_receiver_serial_aperture_receiver_2
OK: emitter_receiver_serial_aperture_receiver_3
OK: emitter_set_channel_emitter
OK: emitter_set_channel_receiver_1
OK: emitter_set_channel_receiver_2
OK: extern
OK: gps
OK: gps_coordinates
OK: gps_coordinates_north_direction
OK: gps_speed
OK: gyro
OK: inertial_unit
OK: joint_end_point_collision_obstacle
OK: joint_end_point_collision_sensor
OK: led
OK: lidar_point_cloud
OK: lidar_rotating
OK: lidar_vs_range_finder
OK: light_sensor_point_spot_light_no_occlusion
OK: motion_loop
OK: motion_regular
OK: motion_reverse
FAILURE with motor: The position value measured by the position sensor should be 1.18212 and not nan
FAILURE with motor2: The max position value measured by the motor 2 should be 1.58 and not nan
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/motor2.wbt: Timeout the results file has not been written quickly enough
OK: pen
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_box.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_box_scaled.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_capsule.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_capsule_scaled.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_cone.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_cone_scaled.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_cylinder.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_cylinder_scaled.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_elevation_grid.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_elevation_grid_scaled.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_indexed_face_set.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_indexed_face_set_scaled.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_plane.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_sphere.wbt: Timeout the results file has not been written quickly enough
FAILURE with /home/travis/build/omichel/webots/tests/api/worlds/pen_sphere_scaled.wbt: Timeout the results file has not been written quickly enough
OK: radar_frustum
OK: radar_occlusion
OK: radar_relative_speed
OK: radar_target_merging
OK: range_finder
OK: range_finder_horizontal_spherical_projection
OK: range_finder_spherical
OK: range_finder_spherical_horizontal_vertical
OK: remote_control
OK: robot_data_supervisor
OK: robot_multiple_step
OK: robot_nested_three_levels
OK: robot_nested_direct_child
OK: robot_nested_group_child
OK: robot_nested_joint_endpoint
OK: robot_nested_slot_endpoint
OK: robot_nested_parent
OK: supervisor_node
OK: robot_synchronous_time
OK: robot_time_consecutive_packets
OK: robot_wait_for_user_input_event
OK: robot_window
OK: robot_window_html
OK: runtime_config_file
OK: supervisor_animation
OK: supervisor_export_image
OK: supervisor_field
OK: supervisor_field_determinism
OK: supervisor_field_import_mf_node
OK: supervisor_get_def
OK: supervisor_get_from_def
OK: supervisor_get_position_orientation
OK: supervisor_get_velocity_with_joints
OK: supervisor_import_remove
OK: supervisor_import_remove_mf
OK: supervisor_import_robot_node
OK: supervisor_inertia
OK: supervisor_internal_field
OK: supervisor_node
OK: supervisor_node_set_visibility
OK: supervisor_reset_physics
OK: supervisor_reset_physics_hinge
OK: supervisor_reset_simulation_iteration_0
OK: supervisor_reset_simulation_iteration_1
OK: supervisor_restart_controller
OK: supervisor_restart_controller_display
OK: supervisor_save_world
OK: supervisor_self_restart
OK: supervisor_set_hinge_position_dynamic
OK: supervisor_set_hinge_position_kinematic
OK: supervisor_set_position_loop
OK: supervisor_set_position_orientation
OK: supervisor_simulation_mode
OK: supervisor_start_stop_movie
OK: supervisor_velocity
OK: touch_sensor_bumper
OK: touch_sensor_force
OK: touch_sensor_force3d
OK: touch_sensor_kinematic
FAILURE with track_caterpillar: Left motor position of tracked robotis wrong after moving straight.
FAILURE with track_invisible: Robot 2 position X after activating the brake is wrong.

### physics test

FAILURE with ball_joint_vs_hinge_joints: Positions of the kinematic and dynamic hinge joints targets differ.
OK: check_saved_hidden_parameter
OK: collision_multiple_trimesh
OK: connector_detach
OK: connector_static_autolock
OK: damping
OK: determinism
OK: dynamic_distance_sensor_rays
OK: dynamic_light_sensor_rays
OK: dynamic_radar_rays
OK: dynamic_receiver_rays
OK: elevation_grid_rotation
OK: floating_point_precision
OK: hinge_2_joint_damping
OK: hinge_joint_damping
OK: hinge_joint_slot
FAILURE with check_joint_hard_limits_after_move: Before moving the robot the SliderJoint hard limit is not respected: position nan > maxStop 0.400.
FAILURE with /home/travis/build/omichel/webots/tests/physics/worlds/joint_hard_limits_after_move.wbt: Timeout the results file has not been written quickly enough
OK: kinematic_geometry_update
OK: move_vehicle_with_hinge_2_joint
OK: move_car
```

I will relaunch it.

**Comment by omichel** [comment]
The test is currently running for the past 24 minutes.

**Comment by DavidMansolino** [comment]
Unfortunately, I can reproduce the exact same failure of the test suite locally.

**Comment by omichel** [comment]
OK, so this PR likely broke something...

**Pull request merged**